data/outport/common.go

```diff
@@ -0,0 +1,61 @@
+package outport
+
+import (
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	"github.com/multiversx/mx-chain-core-go/data"
+	"github.com/multiversx/mx-chain-core-go/data/block"
+	"github.com/multiversx/mx-chain-core-go/marshal"
+)
+
+// GetHeaderBytesAndType returns the marshalled header bytes along with header type, if known
+func GetHeaderBytesAndType(marshaller marshal.Marshalizer, headerHandler data.HeaderHandler) ([]byte, core.HeaderType, error) {
+	if check.IfNil(marshaller) {
+		return nil, "", core.ErrNilMarshalizer
+	}
+
+	var err error
+	var headerBytes []byte
+	var headerType core.HeaderType
+
+	switch header := headerHandler.(type) {
+	case *block.HeaderV2:
+		headerType = core.ShardHeaderV2
+		headerBytes, err = marshaller.Marshal(header)
```

Comment:
L24, L27, L30 can be taken outside of the switch

Reply:
Right, changed as suggested